### PR TITLE
Use error_message function for error handling

### LINKED ISSUE #154: We should use niXXX_error_message, not niXXX_GetErrorMessage
When retrieving errors from the driver in session.py, [we call niXXX_GetErrorMessage](https://github.com/ni/nimi-python/blob/master/build/templates/session.py.mako#L208) (where niXXX is replaced at code-generation with the correct driver prefix).

This works for NI-DMM, but turns out that not all MI drivers support this function - it is an NI-specific extension and not part of the IVI specification. For example, [NI-SWITCH](http://zone.ni.com/reference/en-XX/help/375472A-01/TOC87.htm) doesn't have it.

What all drivers do have is niXXX_error_message, which is equivalent but limits the buffer size for the string to 256 characters (no IVI-dance!), which can lead to truncation if the error message is longer. Using this function for the Python bindings is probably acceptable.

The alternatives are:

- Code generate this part of the bindings differently for each driver (added complexity)
- File NI-internal Corrective Action Requests to drivers that don't support niXXX_GetErrorMessage, wait for them to implement it and ship, then bindings won't work on older drivers. This is likely unacceptable.

### PR DESCRIPTION
**Pull request by Fladolcetta**
[x] This contribution adheres to [CONTRIBUTING.md]

### What does this Pull Request accomplish?

Replaces GetErrorMessage with error_message when getting error descriptions on an invalid session. We do this because it's present in all drivers even though string length is limited.

### List issues fixed by this Pull Request below, if any.

Fixes #154 

### What testing has been done?

Ran System Tests, Unit Tests, Flake8, and then tried to get error_message by doing:
    session1 = nidmm.Session(args.name)
    session2 = nidmm.Session(args.name)
    session1.read()

### CODE REVIEW

File: generated/nidmm/library.py [hunk 1]
```diff
@@ -575,6 +576,14 @@ def niDMM_close(self, vi):  # noqa: N802
                 self.niDMM_close_cfunc.restype = ViStatus_ctype  # noqa: F405
         return self.niDMM_close_cfunc(vi).value
 
+    def niDMM_error_message(self, vi, error_code, error_message):  # noqa: N802
```
Comment (marcoskirsch) [inline generated/nidmm/library.py]: if you add this function to NI-DMM, you should remove the one we no longer use.

File: generated/nifake/tests/test_session.py [hunk 2]
```diff
@@ -477,35 +477,35 @@ def test_get_attribute_error(self):
                 assert e.code == test_error_code
                 assert e.description == test_error_desc
 
-    # Error descriptions
-
-    def test_get_error_description_get_error_message(self):
+    '''
```
Comment (marcoskirsch) [inline generated/nifake/tests/test_session.py]: Why is this commented out?
Add a TODO that explains why and when it needs to be put back in.

File: generated/nifake/tests/test_session.py [hunk 3]
```diff
@@ -477,35 +477,35 @@ def test_get_attribute_error(self):
                 assert e.code == test_error_code
                 assert e.description == test_error_desc
 
-    # Error descriptions
```
Comment (marcoskirsch) [inline generated/nifake/tests/test_session.py]: Put back "# Error descriptions"?
The whole thing is sort of organized with these headers.

File: src/nidmm/metadata/functions_addon.py [hunk 4]
```diff
@@ -12,7 +12,7 @@
     'CheckAttribute.+':                { 'codegen_method': 'no',       },  # We do not include any Check Attribute functions
     '.etAttribute.+':                  { 'codegen_method': 'private',  },  # All Set/Get Attribute functions are private
     'init':                            { 'codegen_method': 'no',       },
-    'error_message':                   { 'codegen_method': 'no',       },
+    'error_message':                   { 'codegen_method': 'private',  },
     'GetError':                        { 'codegen_method': 'private',  },
     'GetErrorMessage':                 { 'codegen_method': 'private',  },
```
Comment (marcoskirsch) [inline src/nidmm/metadata/functions_addon.py]: If these are now dead code, then don't generate.
Applies to all.

File: src/niswitch/metadata/__init__.py [hunk 5]
```diff
@@ -10,12 +10,14 @@
...
 from metadata.functions_addon import functions_params_types
 from metadata.functions_addon import functions_buffer_info
+from metadata.functions_addon import functions_is_error_handling
 from metadata.functions_addon import function_default_value
 
 helper.merge_dicts(functions, functions_codegen_method)
 helper.merge_dicts(functions, functions_enums)
 helper.merge_dicts(functions, functions_params_types)
 helper.merge_dicts(functions, functions_buffer_info)
+helper.merge_dicts(functions, functions_is_error_handling)
```
Comment (marcoskirsch) [inline src/niswitch/metadata/__init__.py]: This is an unrelated fix, isn't it?

File: generated/nidmm/library.py (hunk 1 above)
Comment (Fladolcetta) [reply generated/nidmm/library.py]: Fixed

File: generated/nifake/tests/test_session.py (hunk 3 above)
Comment (Fladolcetta) [reply generated/nifake/tests/test_session.py]: Fixed

File: generated/nifake/tests/test_session.py (hunk 2 above)
Comment (Fladolcetta) [reply generated/nifake/tests/test_session.py]: Fixed

File: src/nidmm/metadata/functions_addon.py (hunk 4 above)
Comment (Fladolcetta) [reply src/nidmm/metadata/functions_addon.py]: fixed

File: src/niswitch/metadata/__init__.py (hunk 5 above)
Comment (Fladolcetta) [reply src/niswitch/metadata/__init__.py]: Sort of? We were not merging this from functions.addon. In order to test the fix to use error_message, this needs to be added

Comment (marcoskirsch) [reply src/niswitch/metadata/__init__.py]: ok
